### ISSUE TITLE
Passing null string in an array of string in service method parameter throws ArgumentNullException

### ISSUE THREAD

**Issue by jbrookssmokeball** [tylerjensen/ServiceWire#50]
If a service method parameter is an array of string, then passing in a null value causes an `ArgumentNullException` to be thrown.

Service method:
```
public interface ITestService
{
    void PassArrayOfString(string[] passedObject);
}
```

Calling the proxy:
```
testServiceProxy.PassArrayOfString(new[] { "abc", "def", null });
```

Throws an `ArgumentNullException`:
```
Message: 
  System.ArgumentNullException : Value cannot be null. (Parameter 'value')

Stack Trace: 
  BinaryWriter.Write(String value)
  ParameterTransferHelper.SendParameters(Boolean useCompression, Int32 compressionThreshold, BinaryWriter writer, Object[] parameters)
  StreamingChannel.InvokeMethod(String metaData, Object[] parameters)
  ITestServiceProxy.PassArrayOfString(String[] )
  TestServiceProxy.PassArrayOfString(String[] passedObject) line 78
```

If the parameter is a `List<string>` then `null` string values in the list serialize okay. It only fails when it's a `string[]` or `params string[]`.

**Comment by jbrookssmokeball** [comment]
Also when returning an array of string
e.g.
```
public interface ITestService
{
    string[] ReturnArrayOfString();
}
```

If you return a null value in the array from the service:
```
public string[] ReturnArrayOfString()
{
    return new[] { "abc", "def", null };
}
```

It throws an `EndOfStreamException` when calling the method over the wire:
```
Message: 
    System.IO.EndOfStreamException : Unable to read beyond the end of the stream.

Stack Trace: 
    BinaryReader.Read7BitEncodedInt()
    BinaryReader.ReadString()
    ParameterTransferHelper.ReceiveParameters(BinaryReader reader)
    StreamingChannel.InvokeMethod(String metaData, Object[] parameters)
    ITestServiceProxy.ReturnArrayOfString()
    TestServiceProxy.ReturnArrayOfString() line 268
```

**Comment by tylerje** [comment]
Fixed in version 5.5.0 being published now.
